process/interceptors/interceptedDataVerifier_test.go

```diff
@@ -93,13 +100,16 @@ func TestInterceptedDataVerifier_CheckValidityShouldNotWork(t *testing.T) {
 
 	err := verifier.Verify(interceptedDataWithErr)
 	require.Equal(t, ErrInvalidInterceptedData, err)
+	require.Equal(t, int64(0), checkValidityCounter.Get())
```

Comment:
looking a bit over the test, this seems like an invalid test, as it returns both an error and no error for the same data hash, and this should not happen.